src/screens/Assistant/SelectFavouritesBottomSheet.tsx

```diff
@@ -0,0 +1,239 @@
+  const {t} = useTranslation();
+  const {
+    favoriteDepartures,
+    frontPageFavouriteDepartures,
+    addFrontPageFavouriteDeparture,
+    removeFrontPageFavouriteDeparture,
+  } = useFavorites();
+  const favouriteItems = favoriteDepartures ?? [];
+  const frontpageFavouriteItems = frontPageFavouriteDepartures ?? [];
+
+  const handleSwitchFlip = (favouriteId: string) => {
+    const favouriteInFavourites = favouriteItems.find(
+      (item) => item.id === favouriteId,
+    );
+
+    const favouriteInFrontpageFavourites = frontpageFavouriteItems.find(
+      (item) => item.id === favouriteId,
+    );
+
+    if (!favouriteInFrontpageFavourites) {
+      const favourite = favoriteDepartures.find(
+        (departure) => departure.id === favouriteId,
+      );
+      if (!favourite) return;
+      addFrontPageFavouriteDeparture(favourite);
+    } else if (favouriteInFavourites) {
+      removeFrontPageFavouriteDeparture(favouriteInFrontpageFavourites);
+    }
+  };
+
+  return (
+    <BottomSheetContainer>
+      <ScreenHeaderWithoutNavigation
+        title={t(SelectFavouriteDeparturesText.header.text)}
+        leftButton={{
+          type: 'cancel',
+          onPress: close,
+          text: 'Close',
+        }}
+        color="background_1"
+      />
+
+      <ScrollView style={styles.flatListArea}>
+        <ThemeText style={styles.questionText} type="heading__component">
+          {t(SelectFavouriteDeparturesText.title.text)}
+        </ThemeText>
+
+        <View>
+          {favouriteItems &&
+            favouriteItems.map((departureDetails, index) => {
+              return (
+                <View key={departureDetails.id}>
+                  <SelectableFavouriteDeparture
+                    handleSwitchFlip={handleSwitchFlip}
+                    favouriteId={departureDetails.id}
+                    active={
+                      frontPageFavouriteDepartures.find(
+                        (departure) => departure.id === departureDetails.id,
+                      )
+                        ? true
+                        : false
+                    }
+                    departureStation={departureDetails.quayName}
+                    lineIdentifier={departureDetails.lineLineNumber ?? ''}
+                    lineName={
+                      departureDetails.lineName ??
+                      t(SelectFavouriteDeparturesText.departures.allVariations)
+                    }
+                    lineTransportationMode={
+                      departureDetails.lineTransportationMode ?? LegMode.BUS
+                    }
+                  />
+                  {favouriteItems.length - 1 !== index && <SectionSeparator />}
+                </View>
+              );
+            })}
+          {!favouriteItems && (
+            <MessageBox type="info">
+              <ThemeText>
+                {t(SelectFavouriteDeparturesText.noFavourites.text)}
+              </ThemeText>
+            </MessageBox>
+          )}
```

Comment:
This one needs a check on length as the default is `[]`, will fix that (hopefully before any of you get to review 😅 

(the same applies for the other check)

Reply:
Should be good now 👌